File: 3216-lexicographically-smallest-string-after-a-swap/3216-lexicographically-smallest-string-after-a-swap.py
class Solution:
    def getSmallestString(self, s: str) -> str:
        n = len(s)
        s = list(s)
        for i in range(1, n):
            if (ord(s[i-1]) - ord('0')) % 2 == (ord(s[i]) - ord('0')) % 2:
                if s[i-1] > s[i]:
                    s[i], s[i-1] = s[i-1], s[i]
                    break
        return ''.join(s)

File: LeetCode/3216-lexicographically-smallest-string-after-a-swap/3216-lexicographically-smallest-string-after-a-swap.java
class Solution {
    public String getSmallestString(String s) {
        StringBuilder sb = new StringBuilder(s);
        int n = sb.length();
        for (int i = 0; i < n - 1; i++) {
            if ((sb.charAt(i) - '0') % 2 == (sb.charAt(i + 1) - '0') % 2) {
                if (sb.charAt(i) > sb.charAt(i + 1)) {
                    char temp = sb.charAt(i);
                    sb.setCharAt(i, sb.charAt(i + 1));
                    sb.setCharAt(i + 1, temp);
                    break;
                }
            }
        }
        return sb.toString();

    }
}